modules/oxide-components/src/main/ts/components/inlinetoolbar/InlineToolbar.tsx

```diff
@@ -85,58 +122,98 @@ const Toolbar: FC<ToolbarProps> = ({
+      }
+    }
+    /* eslint-disable-next-line react-hooks/exhaustive-deps -- toolbarRef/triggerRef are stable ref objects */
+  }, [ isOpen, close ]);
+
   useEffect(() => {
     if (persistent) {
       return;
     }
-
-    const handleClickOutside = (event: MouseEvent) => {
-      if (
-        isOpen &&
-        Type.isNonNullable(toolbarRef.current) &&
-        Type.isNonNullable(triggerRef.current) &&
-        event.target instanceof Node
-      ) {
-        const clickedToolbar = toolbarRef.current.contains(event.target);
-        const clickedTrigger = triggerRef.current.contains(event.target);
-        if (!clickedToolbar && !clickedTrigger) {
-          close();
-        }
-      }
-    };
     document.addEventListener('mousedown', handleClickOutside);
     return () => document.removeEventListener('mousedown', handleClickOutside);
-    /* eslint-disable-next-line react-hooks/exhaustive-deps -- toolbarRef/triggerRef are stable ref objects and don't need to be in deps list. */
-  }, [ isOpen, close, persistent ]);
+  }, [ persistent, handleClickOutside ]);
 
-  const getPosition = (sink: HTMLDivElement, trigger: HTMLDivElement) => {
-    const sinkRect = sink.getBoundingClientRect();
-    const triggerRect = trigger.getBoundingClientRect();
+  const anchorName = useMemo(() => `--${Id.generate('inline-toolbar')}`, []);
 
-    // Calculate trigger's position relative to the sink
-    const top = triggerRect.bottom - sinkRect.top;
-    const left = triggerRect.left - sinkRect.left;
+  useEffect(() => {
+    if (!isOpen || !triggerRef.current || !toolbarRef.current) {
+      return;
+    }
+
+    const trigger = triggerRef.current;
+    const toolbar = toolbarRef.current;
+    const anchorElement = (trigger.firstElementChild as HTMLElement) ?? trigger;
+
+    anchorElement.style.setProperty('anchor-name', anchorName);
+    toolbar.style.setProperty('position-anchor', anchorName);
+
+    const gap = toolbar.ownerDocument?.defaultView
+      ? toolbar.ownerDocument.defaultView.getComputedStyle(toolbar).getPropertyValue('--inline-toolbar-gap') || '6px'
+      : '6px';
+
+    const { isBottom, isCenter } = detectAnchorPosition(anchorElement);
+
+    const topValue = isBottom
+      ? `calc(anchor(${anchorName} top) - ${gap})`
+      : `calc(anchor(${anchorName} bottom) + ${gap})`;
+    const leftValue = isCenter
+      ? `calc(anchor(${anchorName} left) + anchor-size(${anchorName} width) / 2)`
+      : `anchor(${anchorName} left)`;
 
-    return {
-      top: `${top}px`,
-      left: `${left}px`
+    toolbar.style.setProperty('top', topValue);
+    toolbar.style.setProperty('left', leftValue);
+
+    const transform = getTransformValue(isCenter, isBottom);
+    if (transform) {
+      toolbar.style.setProperty('transform', transform);
+    }
+
+    toolbar.style.setProperty('position-try-fallbacks', 'flip-block, flip-inline, flip-block flip-inline');
+
+    return () => {
+      anchorElement.style.removeProperty('anchor-name');
```

Comment:
```suggestion
  anchorElement.style.removeProperty('anchor-name');
  
  Arr.each(['position-anchor', 'top', 'left', 'transform', 'position-try-fallbacks'], (property) => {
    toolbar.style.removeProperty(property);
  });
```

We could iterate over the properties to reduce the repetition.